src/main/java/net/neoforged/neoforge/event/EventHooks.java

```diff
@@ -828,37 +829,65 @@ public static void onLivingConvert(LivingEntity entity, LivingEntity outcome) {
         NeoForge.EVENT_BUS.post(new LivingConversionEvent.Post(entity, outcome));
     }
 
+    @Deprecated(since = "1.21.1", forRemoval = true)
     public static EntityTeleportEvent.TeleportCommand onEntityTeleportCommand(Entity entity, double targetX, double targetY, double targetZ) {
         EntityTeleportEvent.TeleportCommand event = new EntityTeleportEvent.TeleportCommand(entity, targetX, targetY, targetZ);
         NeoForge.EVENT_BUS.post(event);
         return event;
     }
 
+    public static EntityTeleportEvent.TeleportCommand onEntityTeleportCommand(Entity entity, GlobalVec3 target) {
+        EntityTeleportEvent.TeleportCommand event = new EntityTeleportEvent.TeleportCommand(entity, target);
+        NeoForge.EVENT_BUS.post(event);
+        return event;
+    }
+
+    @Deprecated(since = "1.21.1", forRemoval = true)
     public static EntityTeleportEvent.SpreadPlayersCommand onEntityTeleportSpreadPlayersCommand(Entity entity, double targetX, double targetY, double targetZ) {
         EntityTeleportEvent.SpreadPlayersCommand event = new EntityTeleportEvent.SpreadPlayersCommand(entity, targetX, targetY, targetZ);
         NeoForge.EVENT_BUS.post(event);
         return event;
     }
 
+    public static EntityTeleportEvent.SpreadPlayersCommand onEntityTeleportSpreadPlayersCommand(Entity entity, GlobalVec3 target) {
+        EntityTeleportEvent.SpreadPlayersCommand event = new EntityTeleportEvent.SpreadPlayersCommand(entity, target);
+        NeoForge.EVENT_BUS.post(event);
+        return event;
+    }
+
+    @Deprecated(since = "1.21.1", forRemoval = true)
     public static EntityTeleportEvent.EnderEntity onEnderTeleport(LivingEntity entity, double targetX, double targetY, double targetZ) {
         EntityTeleportEvent.EnderEntity event = new EntityTeleportEvent.EnderEntity(entity, targetX, targetY, targetZ);
         NeoForge.EVENT_BUS.post(event);
         return event;
     }
 
+    public static EntityTeleportEvent.EnderEntity onEnderTeleport(LivingEntity entity, GlobalVec3 target) {
+        EntityTeleportEvent.EnderEntity event = new EntityTeleportEvent.EnderEntity(entity, target);
+        NeoForge.EVENT_BUS.post(event);
+        return event;
+    }
+
     @ApiStatus.Internal
-    public static EntityTeleportEvent.EnderPearl onEnderPearlLand(ServerPlayer entity, double targetX, double targetY, double targetZ, ThrownEnderpearl pearlEntity, float attackDamage, HitResult hitResult) {
-        EntityTeleportEvent.EnderPearl event = new EntityTeleportEvent.EnderPearl(entity, targetX, targetY, targetZ, pearlEntity, attackDamage, hitResult);
+    public static EntityTeleportEvent.EnderPearl onEnderPearlLand(ServerPlayer entity, GlobalVec3 target, ThrownEnderpearl pearlEntity, float attackDamage, HitResult hitResult) {
```

Comment:
While it wasn't done by this PR, this method being internal whilst the other methods of its family are not internal strikes me as odd, and an inconsistency. 

We should look into why this specific method is internal, and whether to remove its annotation or apply the same to the rest.

We should also factor in here the decision to mark as internal the constructor of those classes; I think we could leave those constructors unmarked and these marked as internal, but I'm not fully sure yet.

Reply:
I had a look at that and decided to leave it as-is because this one is fired from a very specific vanilla entity and I can see event subscribers assuming it is the only one that can fire it.

And adding Internal to the others would be breaking, so it's off-limits at the moment.